### DISCUSSION TITLE
3.x Allow disabling default authentication routes easily

### DISCUSSION THREAD

**Discussion by realslimsutton** [filamentphp/filament#5356] [Ideas]
Currently in 2.x you can pretty easily disable the default `Login` route by setting the Filament config option to null.
The main problem with this approach is you then also have to customise the auth middleware provided by Filament by creating a new middleware (by extending the default `Authenticate` middleware provided by Filament and updating the `redirectTo` method).

One example is replacing the basic authentication (email/password) with oauth, such as only allowing users to login via Google/Discord/Facebook/etc.

Perhaps with the new upcoming rework of the authentication in 3.x (addition of register, etc), we could make disabling the default authentication routes and easily customising the redirect route for the middleware. Even perhaps generalising the auth routes, so instead of `filament.auth.login`, it's just `auth.login` (or similar), which I believe makes reading the routes easier when say you have users using the same login page as admin users (which means they may not be redirected to the Filament dashboard after logging in).

Unsure if this has already been done in 3.x, but might be worth considering if not.

**Comment by danharrin** [comment]
In 3.x you can pass any "route action". For example, a closure where you return a redirect. Or a controller method.
